src/blocks/CMakeLists.txt

```diff
@@ -26,20 +26,20 @@
 # STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 # THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-if(APPLE)
-    add_library(blocks STATIC blockexports.c)
-    set_target_properties(blocks PROPERTIES LINKER_LANGUAGE C)
-else()
-    if(LINUX_32)
-        add_custom_command(OUTPUT blocks.o MAIN_DEPENDENCY blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} -m elf_i386 ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/blocks.o blocks.dat)
-        add_custom_command(OUTPUT testnet_blocks.o MAIN_DEPENDENCY testnet_blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} -m elf_i386 ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/testnet_blocks.o testnet_blocks.dat)
-        add_custom_command(OUTPUT stagenet_blocks.o MAIN_DEPENDENCY stagenet_blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} -m elf_i386 ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/stagenet_blocks.o stagenet_blocks.dat)
-    else()
-        add_custom_command(OUTPUT blocks.o MAIN_DEPENDENCY blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/blocks.o blocks.dat)
-        add_custom_command(OUTPUT testnet_blocks.o MAIN_DEPENDENCY testnet_blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/testnet_blocks.o testnet_blocks.dat)
-        add_custom_command(OUTPUT stagenet_blocks.o MAIN_DEPENDENCY stagenet_blocks.dat COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR} && ${CMAKE_LINKER} ${LD_RAW_FLAGS} -r -b binary -o ${CMAKE_CURRENT_BINARY_DIR}/stagenet_blocks.o stagenet_blocks.dat)
-    endif()
-    add_library(blocks STATIC blocks.o testnet_blocks.o stagenet_blocks.o blockexports.c)
-    set_target_properties(blocks PROPERTIES LINKER_LANGUAGE C)
-endif()
+set(GENERATED_SOURCES "")
 
+foreach(BLOB_NAME checkpoints testnet_blocks stagenet_blocks)
+    set(OUTPUT_C_SOURCE "generated_${BLOB_NAME}.c")
+    list(APPEND GENERATED_SOURCES ${OUTPUT_C_SOURCE})
+    set(INPUT_DAT_FILE "${BLOB_NAME}.dat")
+    set(GENERATE_C_SOURCE_FROM_BLOB perl -e "\"local$$/\;@blob=unpack('C*',<>)\;print('#include\t<stddef.h>',chr(10),'const\tunsigned\tchar\t${BLOB_NAME}[]={',join(',',@blob),'}\;',chr(10),'const\tsize_t\t${BLOB_NAME}_len=',0+@blob,'\;')\"")
```

Comment:
Is the size variable necessary, shouldn't `sizeof` suffice here? The `epee::span<const unsigned char>` overload constructor should also be able to pickup the correct length using the array length.

Reply:
Will change to `sizeof` if we will come to a consensus on `epee`

Reply:
Or will change to `epee:span<const unsigned char>` otherwise

Reply:
Oh, nope, we can't use `sizeof` here, take a look at https://github.com/monero-project/monero/pull/4472/commits/5490ae6b85f1964250d9aab7a272e87f44226359#diff-13e66019ccd42b6d770225acf8b0f3f2R6